packages/playwright/src/AxeCoreWrapper/DefaultAxeCoreWrapper.cs

```diff
@@ -1,108 +1,297 @@
+        public async Task<AxeResult> RunOnLocator(ILocator locator, AxeRunOptions? options = null)
+        {
+            var frame = await (await locator.ElementHandleAsync()).OwnerFrameAsync();
+            if (frame == null)
+            {
+                throw new Exception("Could not locate frame associated with locator");
+            }
+            return await RunInner(frame, await locator.ElementHandleAsync(), options);
+        }
+
+        /// <inheritdoc/>
+        public async Task<AxeResult> RunLegacy(IPage page, AxeRunContext? context = null, AxeRunOptions? options = null)
+        {
+            await m_axeContentEmbedder.EmbedAxeCoreIntoPage(page, options?.Iframes);
+
+            AxeResult axeResult = await EvaluateAxeRun(page, context, options);
+
+            return axeResult;
+        }
+
+        /// <inheritdoc/>
+        public async Task<AxeResult> RunLegacyOnLocator(ILocator locator, AxeRunOptions? options = null)
+        {
+            await m_axeContentEmbedder.EmbedAxeCoreIntoPage(locator.Page, options?.Iframes);
+
+            string? paramString = JsonConvert.SerializeObject(options);
+            string runParamTemplate = options != null ? "JSON.parse(runOptions)" : string.Empty;
+
+            object jsonObject = await locator.EvaluateAsync<object>($"(node, runOptions) => window.axe.run(node, {runParamTemplate})", paramString);
+            return DeserializeResult(jsonObject);
+        }
+
+        private static async Task<AxeResult> EvaluateAxeRun(IPage page, AxeRunContext? context = null, object? param = null)
+        {
+            string? paramString = JsonConvert.SerializeObject(param);
+
+            string runParamTemplate = param != null ? "JSON.parse(runOptions)" : string.Empty;
+
+            string? contextParam = context is null ? string.Empty : ($"JSON.parse(\'{JsonConvert.SerializeObject(context)}\'),");
+
+            object jsonObject = await page.EvaluateAsync<object>($"(runOptions) => window.axe.run({contextParam}{runParamTemplate})", paramString);
+
+            return DeserializeResult(jsonObject);
+        }
+
+        private static AxeResult DeserializeResult(object jsonObject)
+        {
+            if (jsonObject is null)
+            {
+                throw new Exception();
+            }
+
+            JObject jObject = JObject.FromObject(jsonObject);
+
+            return new AxeResult(jObject);
+        }
+
+        private static List<AxeRuleMetadata> DeserializeRuleMetadata(object jsonObject)
+        {
+            string result = JsonConvert.SerializeObject(jsonObject);
+            var results = JsonConvert.DeserializeObject<List<AxeRuleMetadata>>(result);
+
+            if (results is null)
+            {
+                throw new Exception();
+            }
+
+            return results;
+        }
+
+        /// <summary>
+        /// Runs axe via legacyScan.js at a specific context, which will be passed as-is to Selenium for legacyScan.js to interpret, and

```

Comment:
This comment looks like a copy-paste error - there is no `legacyScan.js` and it references Selenium